tf-build-tools/tf-cmd-api/src/test/java/com/deliveredtechnologies/terraform/terraformhandler/TerraformHandlerTest.java

```diff
@@ -0,0 +1,46 @@
+package com.deliveredtechnologies.terraform.terraformhandler;
+
+import com.deliveredtechnologies.io.Executable;
+import com.deliveredtechnologies.terraform.handler.TerraformHandler;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+import org.slf4j.Logger;
+
+import java.util.Properties;
+
+public class TerraformHandlerTest {
+  private Properties properties;
+  private Executable executable;
+  private Logger logger;
+
+  /**
+   * Sets up the properties, mock(s) and the default terraform directory.
+   */
+  @Before
+  public void setup() {
+    properties = new Properties();
+    executable = Mockito.mock(Executable.class);
+    logger = Mockito.mock(Logger.class);
+  }
+
+  @Test
+  public void executeHandlerWithDoAction() {
```

Comment:
Two scenarios: 
1. nextHandler is present 
   set the next Handler
   call handleRequest
   validate that doAction was called on the current Handler
   validate that doAction was called on the next Handler
2. nextHandler is not present.
   set the next Handler
   call handleRequest
   validate that doAction was called on the current Handler